lib/Driver/Driver.cpp

```diff
@@ -268,6 +268,47 @@ class Driver::InputInfoMap
 };
 using InputInfoMap = Driver::InputInfoMap;
 
+// Get the filename for build record.
+// Additionally set 'outputBuildRecordForModuleOnlyBuild' to true if this is
+// full compilation with swiftmodule.
+static bool getCompilationRecordPath(std::string &buildRecordPath,
```

Comment:
I think this refactoring helps a lot! How about a DOxygen comment for this function, which among other things could explain that it returns true for error and sets buildRecordPath and outputBuildRecordForModuleOnlyBuild?